### ISSUE TITLE
intros! missing whitespace after period for multiple autogenerated intros

### ISSUE THREAD

**Issue by gnuvan** [cpitclaudel/company-coq#23]
When intros! generates several calls to intros, they are separated with a period without whitespace.

Consider the following Coq script (23 True hypotheses imply False):

Goal True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> True -> False.

Now, when doing intros!   here is the generated sequence of tactic calls:

intros H H0 H1 H2 H3 H4 H5 H6 H7 H8 H9 H10 H11 H12 H13 H14 H15 H16 H17 H18 H19 H20.intros H21.

There should be a whitespace after "H20." so that intros can be considered as a tactic call rather than a name qualified by the "H20" namespace.

In other words, there should be a whitespace after each "intros ... ." tactic line generated by intros! for each line output by Show Intros.

(This seemingly small bug can be annoying for autogenerated lemmas with many hypotheses.)

**Comment by cpitclaudel** [comment]
Thanks for the report! That's actually a Proof-General bug (the intros! feature already exists in PG, bound to `C-c C-a TAB`). Could you fill a quick bug report on https://github.com/ProofGeneral/PG with a link to this one? (I moved PG to github last week-end, so bug reporting became a lot easier)

I don't think PG should generate multiple intros, especially since the number that fit on one line is dependent on the printing width setting. For example, the bug disappears if you run `Set Printing Width 100.` before your particular example (of course, with more hypotheses, the same issue would happen again.

I'll probably fix this on the PG side over the week-end :)

**Comment by Matafou** [comment]
Hi,

Clément why not several intros?

**Comment by Matafou** [comment]
you prefer one intros split on several lines? Or one big line?
I have a patch that puts the missing \n (and indent the lines inserted).

**Comment by Matafou** [comment]
I fixed by adding a newline and removing supplementary intros.

**Comment by cpitclaudel** [comment]
Hey Pierre,

No strong opinions, but it felt slightly inconsistent that changing the window width would change the behvaiour of Show Intros. I think the way you fixed it is great. Thanks for reacting faster than I did :)

**Comment by Matafou** [comment]
Maybe the best would probably be to split the line according to fill-column instead of the Printing Width of coq?

**Comment by cpitclaudel** [comment]
That would work as well. I'd expect people who want splitting to run the appropriate M-q or have a mode that does that automatically, though. Thus I'd say any of these solutions (including not breaking the line at all) is acceptable.
